### ISSUE TITLE
sonos if_playing doesn't works with grouped devices

### ISSUE THREAD

**Issue by pzim-devdata** [avantrec/soco-cli#48]
Hello, thank you very much for soco-cli, it's really great :-)

I have found a little problem with `sonos if_playing` command with grouped devices.

When I group sonos's devices together and when I execute this command : 
`sonos connect group one : roam group one : roam_2 group one : roam_3 group one`
`sonos one if_playing volume 20 : connect if_playing rel_vol -6 : roam if_playing volume 13 : roam_2 vol 10 : roam_3 rel_vol 6`
Only the volume of sonos one, roam_2 and roam_3 change

So it's only changing the volume of the coordinator `one` with `ìf_playing`

**Comment by pwt** [comment]
Thanks for the report. Could you run the same command with `--log=info` and send me the output, please? Thanks.

**Comment by pzim-devdata** [comment]
Sure !
`sonos one if_playing volume 20 : connect rel_vol -6 : roam volume 13 : buanderie if_playing volume 10 : cuisine if_playing rel_vol 6 : arrieres if_playing vol 37 : one if_playing sub_gain 3 --log=info` : 

```
2023-02-15 11:35:29,951 sonos.py:130 - main() - Setting up handlers for: [<Signals.SIGINT: 2>, <Signals.SIGTERM: 15>]
2023-02-15 11:35:29,951 sonos.py:244 - main() - No 'SPKR' environment variable set
2023-02-15 11:35:29,951 sonos.py:267 - main() - Found 7 action sequence(s): [['one', 'if_playing', 'volume', '20'], ['connect', 'rel_vol', '-6'], ['roam', 'volume', '13'], ['buanderie', 'if_playing', 'volume', '10'], ['cuisine', 'if_playing', 'rel_vol', '6'], ['arrieres', 'if_playing', 'vol', '37'], ['one', 'if_playing', 'sub_gain', '3']]
2023-02-15 11:35:29,952 utils.py:591 - get_speaker() - Trying direct cache lookup
2023-02-15 11:35:29,952 utils.py:594 - get_speaker() - Trying indirect cache lookup
2023-02-15 11:35:29,952 utils.py:597 - get_speaker() - Trying standard discovery with network scan fallback
2023-02-15 11:35:30,266 utils.py:435 - cache_speakers() - Adding speakers to cache: {SoCo("192.168.2.238"), SoCo("192.168.2.101"), SoCo("192.168.2.128"), SoCo("192.168.2.92"), SoCo("192.168.2.133"), SoCo("192.168.2.108")}
2023-02-15 11:35:30,753 match_speaker_names.py:29 - speaker_name_matches() - Found case-insensitive exact speaker name match for 'one' as 'One'
2023-02-15 11:35:30,753 utils.py:605 - get_speaker() - Successful speaker discovery
2023-02-15 11:35:30,753 sonos.py:483 - main() - Invoking 'run_command' with '<SoCo object at ip 192.168.2.101> if_playing ...'
2023-02-15 11:35:30,773 action_processor.py:1873 - if_stopped_or_playing() - Condition: 'if_playing': Speaker 'One' is in state 'PLAYING'
2023-02-15 11:35:30,773 action_processor.py:1885 - if_stopped_or_playing() - Action invoked: 'One volume 20'
2023-02-15 11:35:30,774 action_processor.py:338 - volume_actions() - Setting volume to 20
2023-02-15 11:35:30,828 api.py:125 - run_command() - Return value: (0, '', '')
2023-02-15 11:35:30,828 utils.py:591 - get_speaker() - Trying direct cache lookup
2023-02-15 11:35:30,829 match_speaker_names.py:58 - speaker_name_matches() - Found partial, any-part-of-name match for 'connect' as 'Sonos Connect'
2023-02-15 11:35:30,829 utils.py:605 - get_speaker() - Successful speaker discovery
2023-02-15 11:35:30,829 sonos.py:483 - main() - Invoking 'run_command' with '<SoCo object at ip 192.168.2.128> rel_vol ...'
2023-02-15 11:35:30,830 action_processor.py:355 - relative_volume() - Adjusting relative volume by -6
2023-02-15 11:35:30,878 api.py:125 - run_command() - Return value: (0, '', '')
2023-02-15 11:35:30,878 utils.py:591 - get_speaker() - Trying direct cache lookup
2023-02-15 11:35:30,879 match_speaker_names.py:58 - speaker_name_matches() - Found partial, any-part-of-name match for 'roam' as 'Sonos Roam'
2023-02-15 11:35:30,879 utils.py:605 - get_speaker() - Successful speaker discovery
2023-02-15 11:35:30,879 sonos.py:483 - main() - Invoking 'run_command' with '<SoCo object at ip 192.168.2.92> volume ...'
2023-02-15 11:35:30,879 action_processor.py:338 - volume_actions() - Setting volume to 13
2023-02-15 11:35:30,898 api.py:125 - run_command() - Return value: (0, '', '')
2023-02-15 11:35:30,899 utils.py:591 - get_speaker() - Trying direct cache lookup
2023-02-15 11:35:30,899 match_speaker_names.py:29 - speaker_name_matches() - Found case-insensitive exact speaker name match for 'buanderie' as 'Buanderie'
2023-02-15 11:35:30,899 utils.py:605 - get_speaker() - Successful speaker discovery
2023-02-15 11:35:30,899 sonos.py:483 - main() - Invoking 'run_command' with '<SoCo object at ip 192.168.2.238> if_playing ...'
2023-02-15 11:35:30,900 action_processor.py:2605 - process_action() - Switching to coordinator speaker 'One'
2023-02-15 11:35:30,971 action_processor.py:1873 - if_stopped_or_playing() - Condition: 'if_playing': Speaker 'One' is in state 'PLAYING'
2023-02-15 11:35:30,971 action_processor.py:1885 - if_stopped_or_playing() - Action invoked: 'One volume 10'
2023-02-15 11:35:30,972 action_processor.py:338 - volume_actions() - Setting volume to 10
2023-02-15 11:35:31,016 api.py:125 - run_command() - Return value: (0, '', '')
2023-02-15 11:35:31,017 utils.py:591 - get_speaker() - Trying direct cache lookup
2023-02-15 11:35:31,017 match_speaker_names.py:29 - speaker_name_matches() - Found case-insensitive exact speaker name match for 'cuisine' as 'Cuisine'
2023-02-15 11:35:31,017 utils.py:605 - get_speaker() - Successful speaker discovery
2023-02-15 11:35:31,017 sonos.py:483 - main() - Invoking 'run_command' with '<SoCo object at ip 192.168.2.108> if_playing ...'
2023-02-15 11:35:31,018 action_processor.py:2605 - process_action() - Switching to coordinator speaker 'One'
2023-02-15 11:35:31,035 action_processor.py:1873 - if_stopped_or_playing() - Condition: 'if_playing': Speaker 'One' is in state 'PLAYING'
2023-02-15 11:35:31,035 action_processor.py:1885 - if_stopped_or_playing() - Action invoked: 'One rel_vol 6'
2023-02-15 11:35:31,035 action_processor.py:355 - relative_volume() - Adjusting relative volume by 6
2023-02-15 11:35:31,076 api.py:125 - run_command() - Return value: (0, '', '')
2023-02-15 11:35:31,076 utils.py:591 - get_speaker() - Trying direct cache lookup
2023-02-15 11:35:31,077 utils.py:594 - get_speaker() - Trying indirect cache lookup
2023-02-15 11:35:31,077 utils.py:597 - get_speaker() - Trying standard discovery with network scan fallback
2023-02-15 11:35:31,078 utils.py:601 - get_speaker() - Trying network scan discovery
2023-02-15 11:35:31,078 utils.py:463 - scan() - Performing full discovery scan with timeout = 1.0s
2023-02-15 11:35:32,207 utils.py:435 - cache_speakers() - Adding speakers to cache: {SoCo("192.168.2.238"), SoCo("192.168.2.101"), SoCo("192.168.2.92"), SoCo("192.168.2.128"), SoCo("192.168.2.133"), SoCo("192.168.2.108")}
2023-02-15 11:35:32,208 utils.py:607 - get_speaker() - Failed to discover speaker
Error: Speaker 'arrieres' not found
2023-02-15 11:35:32,208 utils.py:591 - get_speaker() - Trying direct cache lookup
2023-02-15 11:35:32,209 match_speaker_names.py:29 - speaker_name_matches() - Found case-insensitive exact speaker name match for 'one' as 'One'
2023-02-15 11:35:32,209 utils.py:605 - get_speaker() - Successful speaker discovery
2023-02-15 11:35:32,209 sonos.py:483 - main() - Invoking 'run_command' with '<SoCo object at ip 192.168.2.101> if_playing ...'
2023-02-15 11:35:32,252 action_processor.py:1873 - if_stopped_or_playing() - Condition: 'if_playing': Speaker 'One' is in state 'PLAYING'
2023-02-15 11:35:32,253 action_processor.py:1885 - if_stopped_or_playing() - Action invoked: 'One sub_gain 3'
2023-02-15 11:35:32,321 api.py:125 - run_command() - Return value: (0, '', '')
```

**Comment by pwt** [comment]
I've just released v0.4.57 which I believe fixes the issue you're seeing. Good find.
Let me know how it goes! Thanks.

**Comment by pzim-devdata** [comment]
Thank you so much, it was very fast ! I can confirm it's working perfectly now with the new release :-)
